packages/extension/src/hooks/use-extension/use-end-screen.js

```diff
@@ -0,0 +1,45 @@
+import { useCallback, useReducer } from 'react'
+
+const Actions = {
+  add: 'add',
+  remove: 'remove',
+}
+
+const endScreenReducer = (state, action) => {
+  switch (action.type) {
+    case Actions.add:
+      return [...state, action]
+    case Actions.remove: {
+      const index = state.findIndex(
+        endScreen => endScreen.endScreenType === action.endScreenType
+      )
+      return [...state.slice(0, index), ...state.slice(index + 1)]
+    }
+    default:
+      return state
+  }
+}
+
+export const useEndScreen = key => {
+  const [endScreens, dispatch] = useReducer(endScreenReducer, [])
+
+  const lookupEndScreen = useCallback(type =>
+    endScreens.find(endScreen => endScreen.endScreenType === type)
```

Comment:
lets make a new array reverse before this find, to allow the most recent register to be the first find